src/client/components/hiluk-menu/hiluk-menu.tsx

```diff
@@ -65,6 +66,16 @@ export class HilukMenu extends React.Component<HilukMenuProps, HilukMenuState> {
       >{STRINGS.copySpecificUrl}</li>);
     }
 
+    if (externalView) {
+      const getUrl = new Function('cube', externalView.linkGenerator);
```

Comment:
rather than having a 'cube' argument you should pass in a lot of arguments that will just exist in the context.